edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubeUtils.cs

```diff
@@ -0,0 +1,255 @@
+
+    public static class KubeUtils
+    {
+        const string Alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        const string Numeric = "0123456789";
+        const string AllowedCharsLabelValues = "-._";
+        const string AllowedCharsDns = "-";
+        const string AllowedCharsGeneric = "-.";
+        const int MaxK8SValueLength = 253;
+        const int MaxDnsNameLength = 63;
+        const int MaxLabelValueLength = 63;
+        static readonly HashSet<char> AlphaHashSet;
+        static readonly HashSet<char> AlphaNumericHashSet;
+        static readonly HashSet<char> AllowedLabelsHashSet;
+        static readonly HashSet<char> AllowedDnsHashSet;
+        static readonly HashSet<char> AllowedGenericHashSet;
+
+        static bool IsAlpha(char ch) => AlphaHashSet.Contains(ch);
+
+        static bool IsAlphaNumeric(char ch) => AlphaNumericHashSet.Contains(ch);
+
+        static KubeUtils()
+        {
+            AlphaHashSet = new HashSet<char>();
+            foreach (char c in Alphabet)
+            {
+                AlphaHashSet.Add(c);
+            }
+            AlphaNumericHashSet = new HashSet<char>(AlphaHashSet);
+            foreach (char c in Numeric)
+            {
+                AlphaNumericHashSet.Add(c);
+            }
+            AllowedLabelsHashSet = new HashSet<char>(AlphaNumericHashSet);
+            foreach (char c in AllowedCharsLabelValues)
+            {
+                AllowedLabelsHashSet.Add(c);
+            }
+            AllowedDnsHashSet = new HashSet<char>(AlphaNumericHashSet);
+            foreach (char c in AllowedCharsDns)
+            {
+                AllowedDnsHashSet.Add(c);
+            }
+            AllowedGenericHashSet = new HashSet<char>(AlphaNumericHashSet);
+            foreach (char c in AllowedCharsGeneric)
+            {
+                AllowedGenericHashSet.Add(c);
+            }
+        }
+
+
+        // Valid annotation keys have two segments: an optional prefix and name, separated by a slash (/). 
+        // The name segment is required and must be 63 characters or less, beginning and ending with an 
+        // alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between. 
+        // The prefix is optional. If specified, the prefix must be a DNS subdomain
+        public static string SanitizeAnnotationKey(string key)
+        {
+            if (string.IsNullOrEmpty(key))
+            {
+                throw new InvalidKubernetesNameException($"Key '{key}' is null or empty");
+            }
+
+            char[] annotationSplit = { '/' };
+            string[] keySegments = key.Split(annotationSplit, 2);
+            if (keySegments.Count() == 2)
+            {
+                return SanitizeDNSDomain(keySegments[0]) + "/" + SanitizeNameValue(keySegments[1]);
+            }
+            else
+            {
+                return SanitizeNameValue(key);
```

Comment:
On line 68 you throw an exception if a key is empty, but during sanitization you can end up with empty string. Is it ok for annotation key?